src/phalanx/storage/config.py

```diff
@@ -0,0 +1,240 @@
+        """Load the configuration for an application from disk.
+
+        Parameters
+        ----------
+        name
+            Name of the application.
+
+        Returns
+        -------
+        Application
+            Application data.
+        """
+        base_path = Path.cwd() / "applications" / name
+
+        # Load main values file.
+        values_path = base_path / "values.yaml"
+        if values_path.exists():
+            with values_path.open("r") as fh:
+                values = yaml.safe_load(fh)
+        else:
+            values = {}
+
+        # Load environment-specific values files.
+        environment_values = {}
+        for path in base_path.glob("values-*.yaml"):
+            env_name = path.stem[len("values-") :]
+            with path.open("r") as fh:
+                environment_values[env_name] = yaml.safe_load(fh)
+
+        # Load the secrets configuration.
+        secrets_path = base_path / "secrets.yaml"
+        secrets = []
+        if secrets_path.exists():
+            with secrets_path.open("r") as fh:
+                raw_secrets = yaml.safe_load(fh)
+            for key, raw_config in raw_secrets.items():
+                config = SecretConfig.parse_obj(raw_config)
+                secret = Secret(key=key, application=name, **config.dict())
+                secrets.append(secret)
+
+        # Load the environment-specific secrets configuration.
+        environment_secrets = {}
+        for path in base_path.glob("secrets-*.yaml"):
+            env_name = path.stem[len("secrets-") :]
+            with path.open("r") as fh:
+                raw_secrets = yaml.safe_load(fh)
+            env_secrets = []
+            for key, raw_config in raw_secrets.items():
+                config = SecretConfig.parse_obj(raw_config)
+                secret = Secret(key=key, application=name, **config.dict())
+                env_secrets.append(secret)
+            environment_secrets[env_name] = env_secrets
+
+        # Return the resulting application.
+        return Application(
+            name=name,
+            values=values,
+            environment_values=environment_values,
+            secrets=secrets,
+            environment_secrets=environment_secrets,
+        )
+
+    def _load_environment_config(
+        self, environment_name: str
+    ) -> EnvironmentConfig:
+        """Load the configuration for a Phalanx environment.
+
+        Parameters
+        ----------
+        environment_name
+            Name of the environent.
+
+        Returns
+        -------
+        Environment
+            Loaded environment.
+
+        Raises
+        ------
+        InvalidEnvironmentConfigError
+            Raised if the configuration for an environment is invalid.
+        UnknownEnvironmentError
+            Raised if the named environment has no configuration.
+        """
+        values_name = f"values-{environment_name}.yaml"
+        values_path = Path.cwd() / "environments" / values_name
```

Comment:
This is reasonable for now, but I think for a next iteration we should "discover" the root directory of the phalanx repository and then compute paths within phalanx based on that. This would let us use the phalanx CLI within being at its root.

Reply:
Yup, definitely a good idea and on my to-do list.